### ISSUE TITLE
Can't discover domain network address

### ISSUE THREAD

**Issue by wolf0x** [punk-security/SMBeagle#27]
` .\SMBeagle.exe -D -c out.csv -n 10.16.31.0/24`

**my host have virtualization installed with the 192.168 as private. and 10.16.31.0/24 is my physical adapter IP subnet, however, it couldn't start to scan it. via   .\SMBeagle.exe -D -c out.csv -n 10.16.31.0/24**
1. Performing network discovery...
  discovered 0 private networks and 0 private addresses
    private networks:
    private addresses:
2. Filtering networks and addresses...
  filtered and have 0 private networks and 0 private addresses remaining

**and when I start to run  .\SMBeagle.exe -c out.csv -f, it has below outputs, it still means nothing.**

1. Performing network discovery...
  discovered 6 private networks and 0 private addresses
    private networks:
      fe80::/64
      169.254.0.0/16
      192.168.1.0/24
      192.168.52.0/24
      192.168.183.0/24
      192.168.93.0/24
    private addresses:

**Comment by SimonGurney** [comment]
Thanks for raising the issue, I'll try and reproduce it locally.

**Comment by SimonGurney** [comment]
That is really odd, it works without issue on our test systems with the same cmdline you have provided.

Could you provide more details about the system you are running on?  Is it Win10, Win11 etc... 

``` .\SMBeagle.exe -D -c out.csv -n 10.16.31.0/24 ```

```
        ____              __   _____                      _ __
       / __ \__  ______  / /__/ ___/___  _______  _______(_) /___  __
      / /_/ / / / / __ \/ //_/\__ \/ _ \/ ___/ / / / ___/ / __/ / / /
     / ____/ /_/ / / / / ,<  ___/ /  __/ /__/ /_/ / /  / / /_/ /_/ /
    /_/    \__,_/_/ /_/_/|_|/____/\___/\___/\__,_/_/  /_/\__/\__, /
                                           PRESENTS         /____/

                         -- SMBeagle v1.1.0 --


1. Performing network discovery...
  discovered 1 private networks and 0 private addresses
    private networks:
      10.16.31.0/24
    private addresses:
2. Filtering networks and addresses...
  filtered and have 1 private networks and 0 private addresses remaining
    private networks:
      10.16.31.0/24
3. Probing hosts and scanning networks for SMB port 445...
  scanning is complete and we have 0 hosts with reachable SMB services
4. Probing SMB services for accessible shares...
  probing is complete and we have 0 hosts with accessible shares
    reachabled hosts with accessible SMB shares:
5. Enumerating accessible shares, this can be slow...
  !! Error querying user context.  Failing back to a slower ACL identifier.
      We can also no longer check  if a file is deletable
      It is advisable to set the fast flag and only check the ACLs of one file per directory
5a. Enumerating all subdirectories for known paths
5b. Splitting large directories to optimise caching and to batch output
5c. Enumerating directories
  file enumeration complete, 0 files identified
6. Completing the writes to CSV or elasticsearch (or both)
 -- AUDIT COMPLETE --
```

**Comment by wolf0x** [comment]
Its Win10, and tried to run on physical host and VM inside vmware, both failed and same errors.

**Comment by SimonGurney** [comment]
Hmmm, its really odd as the output at point 1 is doing very little when a network is provided on the command line so its not as is if we are seeing issues with network enumeration etc.

Can you add the "-v" switch and see what if flags as public networks?

**Comment by wolf0x** [comment]
regret to say, -v prints out the same.

`.\SMBeagle.exe -D -v -c out.csv -n 10.16.37.0/24`

1. Performing network discovery...
  discovered 0 private networks and 0 private addresses
    private networks:
    private addresses:
  discovered but will ignore the following 0 public addresses:

**Comment by SimonGurney** [comment]
Sorry for the delay on this, I'm going to produce a debug build if you are happy trying it out.  

It's failing very early in the workflow, before we do anything particularly clever so I'm confused as to what it could be in all honesty.

**Comment by wolf0x** [comment]
waiting for your version, nice to test and believe this project is worth to try.

**Comment by SimonGurney** [comment]
Okay so when I run `.\SMBeagle.exe -D -v -c out.csv -n 10.16.37.0/24` on this debug build I get the following output, could you share yours?

Debug package is here: https://github.com/punk-security/SMBeagle/raw/debug-networks/windows-x64.zip

```PS C:\Users\SimonGurney\Downloads> .\SMBeagle.exe -D -v -c out.csv -n 10.16.37.0/24

        ____              __   _____                      _ __
       / __ \__  ______  / /__/ ___/___  _______  _______(_) /___  __
      / /_/ / / / / __ \/ //_/\__ \/ _ \/ ___/ / / / ___/ / __/ / / /
     / ____/ /_/ / / / / ,<  ___/ /  __/ /__/ /_/ / /  / / /_/ /_/ /
    /_/    \__,_/_/ /_/_/|_|/____/\___/\___/\__,_/_/  /_/\__/\__, /
                                           PRESENTS         /____/

                         -- SMBeagle v1.1.1 --

DEBUG: adding manual network '10.16.37.0/24'
DEBUG:AddNetwork:10.16.37.0/24
DEBUG:AddNetwork:TrimmedNetworksLoop:CurrentNet10.16.37.0/24
DEBUG:Program:ManualNetworks:Net:10.16.37.0/24
1. Performing network discovery...
DEBUG:Program:AllNetworks:Net:10.16.37.0/24
  discovered 1 private networks and 0 private addresses
    private networks:
      10.16.37.0/24
    private addresses:
  discovered but will ignore the following 0 public addresses:
2. Filtering networks and addresses...
  filtered and have 1 private networks and 0 private addresses remaining
    private networks:
      10.16.37.0/24
3. Probing hosts and scanning networks for SMB port 445...
  scanning is complete and we have 0 hosts with reachable SMB services
    reachable hosts:
4. Probing SMB services for accessible shares...
  probing is complete and we have 0 hosts with accessible shares
    reachabled hosts with accessible SMB shares:
    accessible SMB shares:
5. Enumerating accessible shares, this can be slow...
  !! Error querying user context.  Failing back to a slower ACL identifier.
      We can also no longer check  if a file is deletable
      It is advisable to set the fast flag and only check the ACLs of one file per directory
5a. Enumerating all subdirectories for known paths
5b. Splitting large directories to optimise caching and to batch output
5c. Enumerating directories
  file enumeration complete, 0 files identified
6. Completing the writes to CSV or elasticsearch (or both)
 -- AUDIT COMPLETE --
```

**Comment by SimonGurney** [comment]
@wolf0x can you test?

**Comment by wolf0x** [comment]
yes, the debug version output like this:

DEBUG: adding manual network '10.16.81.0/24'
DEBUG:AddNetwork:10.16.81.0/24
DEBUG:AddNetwork:TrimmedNetworksLoop:CurrentNet10.16.81.0/24
DEBUG:Program:ManualNetworks:Net:10.16.81.0/24
1. Performing network discovery...
DEBUG:Program:AllNetworks:Net:10.16.81.0/24
  discovered 0 private networks and 0 private addresses
    private networks:
    private addresses:
  discovered but will ignore the following 0 public addresses:
2. Filtering networks and addresses...
  filtered and have 0 private networks and 0 private addresses remaining

**Comment by SimonGurney** [comment]
Thanks @wolf0x, can you try the new build from the same url?

https://github.com/punk-security/SMBeagle/raw/debug-networks/windows-x64.zip

I've switched the way we lookup private addresses and added a bit more debug output around the private/public network filtering

**Comment by SimonGurney** [comment]
@wolf0x did you get chance to try the new debug build?

**Comment by wolf0x** [comment]
same outputs with the debug version (C94F797DCAC8A0A95349E5209C9FFD060AB3310AAF64699C75B71973A3FDF5AF)

**Comment by SimonGurney** [comment]
The 10 network you are using is a private network though, which is why it's weird that it's being detected as public.  Is it still thinking 10.26.81.0/24 is public or are you trying to scan another subnet?

**Comment by wolf0x** [comment]
actually, 10 is just a dummy subnets, we are using other segment, some of segments is public not private, so are you able to add a switch to force perform the scan via dedicated segment instead of checking its private/public?

**Comment by SimonGurney** [comment]
Ah, I thought you were sending mr the exact output.  I didn't realise you were changing the addresses before sending it.

I'll add a fix so that manual addresses are scanned regardless of private/public.

**Comment by SimonGurney** [comment]
Hey @wolf0x, can you try this latest build?  https://github.com/punk-security/SMBeagle/raw/debug-networks/windows-x64.zip

I've excluded manual nets and hosts from filtering.

**Comment by wolf0x** [comment]
yup, this time, it list out the alive IPs in the subnet and can run to perform SMB enumeration, however, it seems only performed scan on itself and enumerate admin$, C$ and so on, thus generate a 42MB csv file for the local IP.

**Comment by SimonGurney** [comment]
Can you try with the switches:

-n <network> -D -c out.csv


-n will allow you to feed in the public network and bypass filtering.

-D will disable discovery all together

You can also use -L to disable local share scanning, but you shouldn't need that if discovery is disabled.

There is also a hard coded 500ms maximum wait on the port scan, so scanning targets in different countries may not work.

**Comment by SimonGurney** [comment]
This must be a permission issue, I've just tested against a public IP in Azure and it worked fine, found the test share and file:

If they are domain joined, it should work fine if you have permissions.  If they are not domain joined, make sure the user you are running SMBeagle as exists on the target system and the password is the same (to allow ntlm pass the hash magic to work).

```
c:\windows-x64>SMBeagle.exe -c out.csv -D -n 51.142.76.87/32

        ____              __   _____                      _ __
       / __ \__  ______  / /__/ ___/___  _______  _______(_) /___  __
      / /_/ / / / / __ \/ //_/\__ \/ _ \/ ___/ / / / ___/ / __/ / / /
     / ____/ /_/ / / / / ,<  ___/ /  __/ /__/ /_/ / /  / / /_/ /_/ /
    /_/    \__,_/_/ /_/_/|_|/____/\___/\___/\__,_/_/  /_/\__/\__, /
                                           PRESENTS         /____/

                         -- SMBeagle v1.1.1 --


1. Skipping network discovery due to -D switch...
2. Skipping filtering as network discovery disabled...
3. Processing manual networks and addresses...
  added network '51.142.76.87/32'
4. Probing hosts and scanning networks for SMB port 445...
  scanning is complete and we have 1 hosts with reachable SMB services
5. Probing SMB services for accessible shares...
  probing is complete and we have 1 hosts with accessible shares
    reachabled hosts with accessible SMB shares:
      51.142.76.87
6. Enumerating accessible shares, this can be slow...
6a. Enumerating all subdirectories for known paths
6b. Splitting large directories to optimise caching and to batch output
6c. Enumerating directories
  file enumeration complete, 1 files identified
7. Completing the writes to CSV or elasticsearch (or both)
 -- AUDIT COMPLETE --
```

**Comment by SimonGurney** [comment]
Fixed in #29
